stellar-multitenant/pkg/tenant/manager.go

```diff
@@ -236,6 +238,36 @@ func (m *Manager) SoftDeleteTenantByID(ctx context.Context, tenantID string) (*T
 	return &t, nil
 }
 
+// DeactivateTenantDistributionAccount sets a distribution account of status ACTIVE to PENDING_USER_ACTIVATION for the given tenant id,
+// and is only used in the case where the distribution account is of type CircleDBVault.
+func (m *Manager) DeactivateTenantDistributionAccount(ctx context.Context, tenantID string) error {
+	updateQuery := `
+		UPDATE tenants t
+		SET
+			distribution_account_status = 'PENDING_USER_ACTIVATION'
+	`
+
+	queryParams := &QueryParams{
+		Filters: excludeInactiveTenantsFilters(),
+	}
+	queryParams.Filters[FilterKeyID] = tenantID
+	queryParams.Filters[FilterKeyDistributionAccountType] = schema.DistributionAccountCircleDBVault
+	queryParams.Filters[FilterKeyDistributionAccountStatus] = schema.AccountStatusActive
+
+	query, params := m.newManagerQuery(updateQuery, queryParams)
+	query += " RETURNING *"
+
+	var t Tenant
+	if err := m.db.GetContext(ctx, &t, query, params...); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return ErrTenantDoesNotExist
```

Comment:
This is not accurate. There can be `ErrNoRows` when the Tenant Exists but is already deactivated, or exists and is not Circle, etc. We should still throw an error, but we can just say that the tenant was not deactivated.

Reply:
Good point, IMO we don't need to throw Not sure we need to throw this error when the status is not Active.

Reply:
We don't need to throw an error, but if I follow the logic correctly, we should never get in the case where we attempt deactivating an account that's already deactivated. 

I'm fine with not throwing an error though. It makes the operation idempotent.

Reply:
so to summarize
- we should hard code all the filters in the query params in the sql statement itself
- don’t need to return an error indicating whether a tenant was updated at all, making the operation just idempotent